src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs

```diff
@@ -120,8 +120,8 @@ internal class Descriptors
 
         public static DiagnosticDescriptor AZC0018 = new DiagnosticDescriptor(
             nameof(AZC0018),
-            "Do ensure protocol method take a RequestContext parameter called context and not take models as parameter type or return type.",
-            "Protocol method should have requestContext as the last parameter and don't have model as parameter type or return type.",
+            "Do ensure protocol methods take a RequestContext parameter called `context` and do not take models as parameter or return types. Do ensure protocol methods will not cause an ambiguous overload resolution with convenience methods.",
+            "Protocol method should have `requestContext` as the last parameter and not use a model as parameter or return types. Protocol methods should not have optional parameters if ambiguity exists between the protocol method and convenience methods.",

```

Comment:
> Protocol methods should not have optional parameters if ambiguity exists between the protocol method and convenience methods.

Would it make sense to say that `context` must be required if the convenience method creates an ambiguous overload?

Reply:
If `context` is required then every parameters should be required. I think it would be more specific to say all the parameters should be required.

Reply:
Nit: it might be a little easier to understand to phrase the last sentence as "If ambiguity exists between the protocol method and the convenience method, all parameters of the protocol method should be required." (Instead of "Protocol methods should not have optional parameters if ambiguity exists between the protocol method and convenience methods.")

Reply:
I move this sentence to AZC0019, since AZC0019 is used to detect ambiguity.